extension/content.js

```diff
@@ -170,6 +170,27 @@ function addBlameParentLinks() {
 	});
 }
 
+function addEditButton() {
+	const dirListItems = $('tr.js-navigation-item .js-directory-link');
+	const readmeContainer = $('#readme');
+	let editHref;
+
+	dirListItems.each((i, el) => {
+		const isReadme = /(readme\.\w+)|(readme)$/i.test(el.href);
+		if (isReadme) {
+			editHref = el.href.replace('blob', 'edit');
+		}
+	});
```

Comment:
I still think that would be cleaner. You can easily get the master branch `$('#branch').attr('value')`.